spec/oauth2/access_token_spec.rb

```diff
@@ -75,6 +75,27 @@ def assert_initialized_token(target) # rubocop:disable Metrics/AbcSize
       assert_initialized_token(target)
       expect(target.expires_at).to be_a(Integer)
     end
+
+    describe 'expires_latency' do
+      it 'sets it via options' do
+        hash = {:access_token => token, :expires_latency => '10'}
+        target = described_class.from_hash(client, hash)
+        expect(target.expires_latency).to eq 10
+      end
+
+      it 'sets it 0 by default' do
+        hash = {:access_token => token, :expires_in => 100}
+        target = described_class.from_hash(client, hash)
+        expect(target.expires_latency).to eq 0
+      end
+
+      it 'reduces expires_at by the given amount' do
+        allow(Time).to receive(:now).and_return(1_530_000_000)
+        hash = {:access_token => token, :expires_latency => 10, :expires_in => 100}
+        target = described_class.from_hash(client, hash)
+        expect(target.expires_at).to eq 1_530_000_090
+      end
```

Comment:
The code as you have it would fail a spec where `expires_at` and `expires_latency` are both provided explicitly, as `expires_at` would not be adjusted by the `expires_latency`.  See my comment inline with the code for a suggestion at a small change to make `expires_latency` globally applicable when provided.

Reply:
- [x] Added a spec for this scenario